modpack_format/modpack_example.modip.json

```diff
@@ -0,0 +1,35 @@
+{
+  "formatType": "modipModpack",
+  "formatVersion": "1.0.0",
```

Comment:
Are these for auto-detection? What should the launcher do with them? Is the formatVersion necessary or can it be inferred based on the presence of various fields?

Reply:
`formatType` is there in case we have other format types. For example the regular index schema (what we have in the repo now) would maybe become a separate format type because it's different. `formatVersion` is there just for convenience, comparing based on presence requires more effort and may not work 100% of the time (e.g. optional fields).

Reply:
If version 2 adds a new optional field, why is version 2 without the optional field different from version 1 without the optional field?

Reply:
There may still be edge cases where you cannot determine the format version but they should be parsed differently. Either way it is a convenience to list the format version here. Auto detection is unnecessarily complex IMO.

Reply:
Won't peterix say no, this is too complex, I'm not adding a database of your format versions?

Reply:
I don't think it's too complex. The format should be able to be changed without breaking everything and causing conflicts. Besides, I don't think new format versions will be released very often.

Reply:
What should a launcher do if these fields don't say `modipModpack` and `1.0.0`?

Reply:
If they don't say `modipModpack` or `1.0.0` then an error needs to be thrown because that shouldn't happen. Ideally a message would be shown to the user saying something like "Invalid modpack metadata"

Reply:
If it says 1.0.1 or 0.9.9, is that an error?

Reply:
Probably, yeah. For 0.9.9 that is definitely an error (because the spec follows SemVer). For 1.0.1, I think it would be an error as well but I suppose it could go either way, assuming there are no breaking changes in 1.0.1 - but I'm leaning towards 1.0.1 being considered an error.

Reply:
Then there is no reason for this to be a 3-part string instead of just a number?

Reply:
I suppose it could just be a number. It's currently the semver string because that's what the index spec uses, and I felt it fit, but I can change it to a number if that's better.

Reply:
Maybe the same thing applies to the index spec

Reply:
I've switch to using a number, I think it's better.